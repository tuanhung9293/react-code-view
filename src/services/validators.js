export function required(message = 'This field is required') {
  return value => (
    !value || value.length === 0
  ) && message
}
