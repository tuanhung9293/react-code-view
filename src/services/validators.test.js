import {required} from './validators'

describe('required', () => {
  it('takes a message and returns error message on falsy values', () => {
    const validate = required('test message')
    expect(validate('')).toEqual('test message')
    expect(validate(null)).toEqual('test message')
    expect(validate(undefined)).toEqual('test message')
    expect(validate(false)).toEqual('test message')
  })

  it('returns false if value is truthy', () => {
    const validate = required('test message')
    expect(validate('hello')).toEqual(false)
  })

  it('provides a default message', () => {
    const validate = required()
    expect(validate('')).toMatchSnapshot()
  })
})
