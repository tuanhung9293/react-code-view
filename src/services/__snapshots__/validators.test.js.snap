// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`required provides a default message 1`] = `"This field is required"`;
